### PR TITLE
#136 - Generalize decomposition into general block structures

### PR DESCRIPTION
**Pull request by schillic**
See #136.

### CODE REVIEW

**Comment by schillic** [comment]
While I was doing this, I realized: We could implement *all* project-approximate methods as follows:
```julia
M = sparse([1, 2], [block[1], block[2]], [one(N), one(N)], 2, n)
return overapproximate(M * S, ɛ)
```
This is the implementation for epsilon-close polygon approximation, but it should also work for other types.

Advantage: No extra code in `decompositions.jl` for new set types is needed. All the stuff happens in `overapproximate.jl`.

File: src/Approximations/decompositions.jl [hunk 1]
```diff
@@ -1,145 +1,176 @@
...
-    pn = σ(sparsevec([2*bi], [one(N)], n), S)
-    pw = σ(sparsevec([2*bi-1], [-one(N)], n), S)
-    ps = σ(sparsevec([2*bi], [-one(N)], n), S)
-    return (pe, pn, pw, ps)
-end
 
 """
-    decompose(S::LazySet{N}, ɛi::Vector{<:Real})::CartesianProductArray where {N<:Real}
+    function project(S::LazySet{N},
+                     block::AbstractVector{Int},
```
Comment (mforets) [inline src/Approximations/decompositions.jl]: call it `blocks`? (same below)

Comment (mforets) [reply src/Approximations/decompositions.jl]: ok i see, this is a block of size 2.

**Comment by mforets** [comment]
-> add new functions to Approximations docs

**Comment by mforets** [comment]
test pass and docs build here.

**Pull request merged**